ZConfig/info.py

```diff
@@ -64,13 +64,15 @@ class BaseInfo(object):
 
     def __init__(self, name, datatype, minOccurs, maxOccurs, handler,
                  attribute):
-        if maxOccurs is not None:
-            if maxOccurs < 1:
-                raise ZConfig.SchemaError(
-                    "maxOccurs must be at least 1")
-            if minOccurs is not None and minOccurs > maxOccurs:
-                raise ZConfig.SchemaError(
-                    "minOccurs cannot be more than maxOccurs")
+        assert maxOccurs is not None, "Use Unbounded for an upper bound, not None"
+        assert minOccurs is not None, "Use 0 for a lower bound, not None"
```

Comment:
Sure, and it should do that. This is production code, the assert is there as executable documentation, not a guarantee. It exists to help inform developers of problems (development, run, read, or test time),  not for any sort of sanity checking. If you violate the assertion contract, *there will* be other problems, albeit maybe not as obvious if assertions are disabled. In all these cases, they would be very obvious TypeErrors.